From 46962493a5ee4c410b59e8c5fd59e86f8596bd1a Mon Sep 17 00:00:00 2001
From: PyScotch <pyscotch@noreply.invalid>
Date: Tue, 13 Oct 2026 16:11:12 +0000
Subject: [PATCH 225/240] Bugfix: test overlap partition statistics in CMake
 test `gpart_overlap_2` [report R. Perry]

Unlike its Makefile counterpart, the CMake test ran "gmtst" without
the "-o" flag on an overlap partition, so that the mapping was read as
a plain one and the overlap code path of "gmtst" was not exercised.

Refs: BLD-05
---
 src/check/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/check/CMakeLists.txt b/src/check/CMakeLists.txt
index 66be8b8..db172d2 100644
--- a/src/check/CMakeLists.txt
+++ b/src/check/CMakeLists.txt
@@ -743,7 +743,7 @@ add_test(NAME gpart_overlap_1 COMMAND ${BASH} -c "'$<TARGET_FILE:gpart>' -o 9 '$
   '$<TARGET_FILE:gmtst>' -o '${dat}/bump_b100000.grf' '${tgt}/k9.tgt' bump_b100000_part_ovl_9.map")
 
 add_test(NAME gpart_overlap_2 COMMAND ${BASH} -c "'$<TARGET_FILE:gpart>' -o 32 '${dat}/bump_imbal_32.grf' bump_imbal_32_part_ovl_32.map -vmt && \
-  '$<TARGET_FILE:gmtst>' '${dat}/bump_imbal_32.grf' '${tgt}/k32.tgt' bump_imbal_32_part_ovl_32.map")
+  '$<TARGET_FILE:gmtst>' -o '${dat}/bump_imbal_32.grf' '${tgt}/k32.tgt' bump_imbal_32_part_ovl_32.map")
 
 add_test(NAME gpart_overlap_3 COMMAND ${BASH} -c "'$<TARGET_FILE:gpart>' -o 6 '${dat}/bump_imbal_32.grf' bump_imbal_32_part_ovl_32_6.map '-me{strat=m{asc=b{width=3,bnd=d{pass=40,dif=1,rem=0}f{move=80,pass=-1,bal=0.01},org=f{move=80,pass=-1,bal=0.01}},low=r{job=t,bal=0.01,map=t,poli=S,sep=(m{asc=b{bnd=f{move=120,pass=-1,bal=0.01,type=b},org=f{move=120,pass=-1,bal=0.01,type=b},width=3},low=h{pass=10}f{move=120,pass=-1,bal=0.01,type=b},vert=120,rat=0.8}|m{asc=b{bnd=f{move=120,pass=-1,bal=0.01,type=b},org=f{move=120,pass=-1,bal=0.01,type=b},width=3},low=h{pass=10}f{move=120,pass=-1,bal=0.01,type=b},vert=120,rat=0.8})},vert=10000,rat=0.8,type=h}}f' -vmt")
 
-- 
2.47.3

